ee/desktop/runner/runner.go

```diff
@@ -117,16 +123,23 @@ type processRecord struct {
 	socketPath string
 }
 
+// desktopUserStatus is all the device data sent to the desktop user process
+type DesktopUserStatus struct {
+	Enabled bool          `json:"enabled"`
+	Menu    menu.MenuData `json:"menu"`
+}
+
 // New creates and returns a new DesktopUsersProcessesRunner runner and initializes all required fields
 func New(opts ...desktopUsersProcessesRunnerOption) *DesktopUsersProcessesRunner {
 	runner := &DesktopUsersProcessesRunner{
-		logger:           log.NewNopLogger(),
-		interrupt:        make(chan struct{}),
-		uidProcs:         make(map[string]processRecord),
-		updateInterval:   time.Second * 5,
-		procsWg:          &sync.WaitGroup{},
-		interruptTimeout: time.Second * 10,
-		usersFilesRoot:   filepath.Join(os.TempDir(), "kolide-desktop"),
+		logger:                 log.NewNopLogger(),
+		interrupt:              make(chan struct{}),
+		uidProcs:               make(map[string]processRecord),
+		updateInterval:         time.Second * 5,
+		procsWg:                &sync.WaitGroup{},
+		interruptTimeout:       time.Second * 10,
+		usersFilesRoot:         filepath.Join(os.TempDir(), "kolide-desktop"),
+		processSpawningEnabled: true,
```

Comment:
`processSpawningEnabled` controls whether or not desktop user processes are created, if the desktop runner is disabled all together (like we have it in prod right now), this doesn't matter since the whole desktop subsystem isn't enabled.

Oh.. I think I know what you mean... when we remove this line https://github.com/kolide/launcher/blob/c844b174dd3bbcf77d36061f6812cef9019dadaf/cmd/launcher/launcher.go#L184 we want `processSpawningEnabled`  to default to `false` (desktop disabled) unless explicitly overriden by the control server. I can make that change, but that would also force anyone not talking to a K2 with control endpoints to not get desktop at all (unless I hardcode in defaulting to `true` if `opts.KolideServerURL == "k2device-preprod.kolide.com" || opts.KolideServerURL == "localhost:3443" `)

Would we want a separate bool to control whether or not the menu is displayed? Not sure if the scenario of having desktop enabled, but menu disabled, is something we care to support. Can't do much with the desktop user process without the menu, so that's why I didn't separate those concepts.

Reply:
I don't think we need separate control over the desktop process and the visible menu. Today, the desktop process does nothing other than show a menu. And while maybe that changes in the future, everything changes. 

> we want processSpawningEnabled to default to false (desktop disabled) unless explicitly overriden by the control server

Yes, I think that change is correct. With the caveat that the setting should persist across restarts. (I expect the control server to tell clients to enable desktop, and that it is unlikely to ever be disabled)